src/Server.kt

```diff
@@ -197,16 +199,39 @@ fun Application.module() {
 
                     if (call.parameters.contains("simulationIndex")) {
                         val simIndex = call.parameters["simulationIndex"]?.toInt() ?: 0
-                        call.respondText(model.run(simIndex))
+                        call.respondText(model.run(simIndex, null))
                     } else {
-                        call.respondText(model.run(0))
+                        call.respondText(model.run(0, null))
                     }
                 } catch (err: IndexOutOfBoundsException) {
                     call.respond(HttpStatusCode.NotFound)
                 }
             }
 
         }
+
+        post("/execute/{i}") {
```

Comment:
I understand now that the new post execute method is for running models with the posted simulation and the previous GET execute method is for running the existing simulation at a passed index. We need to write this in the documentation or comments.